sdk/servicebus/Azure.Messaging.ServiceBus/README.md

```diff
@@ -1,104 +1,475 @@
+await receiver.SetSessionStateAsync(Encoding.Default.GetBytes("some state"));
+
+// the state can be retrieved for the session as well
+byte[] state = await receiver.GetSessionStateAsync();
+```
+
+### Using the Processor
+
+```C# Snippet:ServiceBusProcessMessages
+string connectionString = "<connection_string>";
+string queueName = "<queue_name>";
+// since ServiceBusClient implements IAsyncDisposable we create it with "await using"
+await using var client = new ServiceBusClient(connectionString);
+
+// get the sender
+ServiceBusSender sender = client.GetSender(queueName);
+
+// create a message batch that we can send
+ServiceBusMessageBatch messageBatch = await sender.CreateBatchAsync();
+messageBatch.TryAdd(new ServiceBusMessage(Encoding.UTF8.GetBytes("First")));
+messageBatch.TryAdd(new ServiceBusMessage(Encoding.UTF8.GetBytes("Second")));
+
+// send the message batch
+await sender.SendBatchAsync(messageBatch);
+
+// get the options to use for configuring the processor
+var options = new ServiceBusProcessorOptions
+{
+    // By default after the message handler returns, the processor will complete the message
+    // If I want more fine-grained control over settlement, I can set this to false.
+    AutoComplete = false,
+
+    // I can also allow for multi-threading
+    MaxConcurrentCalls = 2
+};
+
+// get a processor that we can use to process the messages
+ServiceBusProcessor processor = client.GetProcessor(queueName, options);
+
+// since the message handler will run in a background thread, in order to prevent
+// this sample from terminating immediately, we can use a task completion source that
+// we complete from within the message handler.
+TaskCompletionSource<bool> tcs = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+processor.ProcessMessageAsync += MessageHandler;
+processor.ProcessErrorAsync += ErrorHandler;
+
+async Task MessageHandler(ProcessMessageEventArgs args)
+{
+    string body = Encoding.Default.GetString(args.Message.Body.ToArray());
+    Console.WriteLine(body);
+
+    // we can evaluate application logic and use that to determine how to settle the message.
+    await args.CompleteAsync(args.Message);
+    tcs.SetResult(true);
+}
+
+Task ErrorHandler(ProcessErrorEventArgs args)
+{
+    // the error source tells me at what point in the processing an error occurred
+    Console.WriteLine(args.ErrorSource);
+    // the fully qualified namespace is available
+    Console.WriteLine(args.FullyQualifiedNamespace);
+    // as well as the entity path
+    Console.WriteLine(args.EntityPath);
+    Console.WriteLine(args.Exception.ToString());
+    return Task.CompletedTask;
+}
+await processor.StartProcessingAsync();
+
+// await our task completion source task so that the message handler will be invoked at least once.
+await tcs.Task;
+
+// stop processing once the task completion source was completed.
+await processor.StopProcessingAsync();
+```
+
+### Working with Sessions
```

Comment:
Can we rename this to something like `Sending and receiving messages from a specific session`? Also can we move this section before `Using the Processor`?

Reply:
I added this afterwards just because I think it is more of a specialized/advanced use case..

Reply:
Yes, but if you see the current sequence it's like:

* Sending and receiving a message
* Sending and receiving a batch of messages
* Complete a message
* Abandon a message
* Defer a message
* Dead letter a message
* Sending and receiving session messages
* Using the Processor
* `Working with Sessions` - This can go after `Sending and receiving session messages` because it is not using processor. Or we can move all the session related stuff after the processor.
* Using the Processor with Sessions

Reply:
Or can we remove `Working with Sessions` example from README?

Reply:
Yeah let's remove the sessions stuff from readme for now.

Reply:
or at least the processor session one.

Reply:
How about we just keep `Sending and receiving session messages` example and remove other session related example?